app/Http/Controllers/Api/PostController.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Models\Post;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Http\Resources\PostResource;
+use App\Http\Requests\User\Posts\StoreRequest;
+
+class PostController extends Controller
+{
+    //Retrieve all Posts
+    public function index(){
+        return PostResource::collection(Post::all());
+    }
+
+    //Store Post
+    public function store(StoreRequest $request){
+        $data = $request -> validated();
+            $post = Post::create($data);
+            if($post){
+                return response()-> json([
+                    'status'=>200,
+                    'message'=> 'Post Created Successfully'
+                ], 200);
+            }else{
+                return response()-> json([
+                    'status'=>500,
+                    'message'=> 'Post Not Successful'
+                ], 500);
+            }
+     }
+
+    //Retrieve Specific Post
+    public function edit( $id){
```

Comment:
Please use Implicit Binding like the other PostController for web.
https://laravel.com/docs/10.x/routing#implicit-binding